File: LeetCode/1701. Average Waiting Time/1701.cs
public class Solution
{
    public double AverageWaitingTime(int[][] customers)
    {
        double totalWaitTime = 0;
        double currentTime = 0;
        foreach (var customer in customers)
        {
            if (currentTime < customer[0])
            {
                currentTime = customer[0];
            }
            currentTime += customer[1];
            totalWaitTime += currentTime - customer[0];
        }
        return totalWaitTime / customers.Length;

    }
}
//other people's solution

/*
public class Solution {
    public double AverageWaitingTime(int[][] customers) {
       int curtime = customers[0][0]; 
       double wSum = 0;
       int waittime = 0;
       for(int i = 0; i < customers.Length; i++){
            if (curtime < customers[i][0]) {
                curtime = customers[i][0];
            }
            waittime = (curtime - customers[i][0]) + customers[i][1];
            curtime = curtime + customers[i][1];
            wSum = wSum + waittime;
           
       }
       return (double)wSum/customers.Length;
    }
}
*/
/*
public class Solution {
    public double AverageWaitingTime(int[][] customers) {
       var d = customers.Length;
        var sum = 0d;
        var chefTime = customers[0][0];
        foreach (var arr in customers)
        {
            var arrTime = arr[0];
            sum += arr[1];
            if (chefTime - arrTime >= 0)
                sum += chefTime - arrTime;
            else
            {
                chefTime = arrTime;
            }
            chefTime += arr[1];
        }
        return sum / d;
    }
}
*/

File: LeetCode/1701. Average Waiting Time/1701.cpp
class Solution
{
public:
    double averageWaitingTime(vector<vector<int>> &customers)
    {
        double waitTime = 0;
        double currentTime = 0;
        for (int i = 0; i < customers.size(); i++)
        {
            currentTime = max(currentTime, (double)customers[i][0]);
            currentTime += customers[i][1];
            waitTime += currentTime - customers[i][0];
        }
        return waitTime / customers.size();
    }
};
//other people's solution
/*
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        ios_base::sync_with_stdio(0);cin.tie(0);cout.tie(0);
        double ans = 0 ;
        int prev = 0 ;
        for(int i =  0; i < customers.size();i++){
            if(prev <= customers[i][0]){
               ans += customers[i][1] ; prev = customers[i][1]  +  customers[i][0] ;
            }
            else{
                ans +=( prev -  customers[i][0] ) +   (customers[i][1])  ;
                prev= prev + (customers[i][1]) ;
            }

        }
        return ans /customers.size() ;
    }
};
*/
/*
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double count = 0, curr = 0;
        for(auto& a : customers)
        {
            curr = max(curr, a[0] * 1.0) + a[1];
            count += curr - a[0];
        }
        return count / customers.size();
    }
};
*/
/*
class Solution {
public:
    double averageWaitingTime(vector<vector<int>>& customers) {
        double n = customers.size();
        double waitTime = 0;
        double timeTaken = customers[0][0]; //this should start with the first customer's arrival

        for(auto cust : customers){
            int arrival = cust[0];
            int prepTime = cust[1];

            if(arrival > timeTaken){
                //chef is free
                timeTaken = arrival + prepTime;
                waitTime += prepTime;
            }else{
                //chef is busy preparing orders
                timeTaken += prepTime;
                waitTime += timeTaken - arrival;
            }
        }

        return waitTime/n;
    }
};
*/